src/compiler/parser.ts

```diff
@@ -4147,7 +4147,7 @@ namespace ts {
             return parseOptional(SyntaxKind.EqualsToken) ? parseAssignmentExpressionOrHigher() : undefined;
         }
 
-        function parseAssignmentExpressionOrHigher(): Expression {
+        function parseAssignmentExpressionOrHigher(parsingConditionalTrueSide = false): Expression {

```

Comment:
I am not attached to this parameter name. Please suggest something better if you think of one!